helm-prereqs/setup.sh

```
@@ -430,6 +505,385 @@ until ROLE_ID_B64="$(kubectl get secret nico-vault-approle-tokens \
+    kubectl rollout status statefulset -n dpf-operator-system \
+        -l app.kubernetes.io/name=argocd-application-controller --timeout=300s
+
+    # 5b.2 Pull + repo secrets. Idempotent (apply of a dry-run render).
+    #      The dpu.nvidia.com/image-pull-secret label makes DPF propagate the
+    #      Secret into DPUService image-pull secrets on the DPU cluster.
+    if [[ -n "${NICO_DPF_NGC_API_KEY}" ]]; then
+        echo "Creating DPF pull and Argo CD repository secrets..."
+        kubectl create secret docker-registry dpf-pull-secret \
+            --namespace dpf-operator-system \
+            --docker-server=nvcr.io \
+            --docker-username='$oauthtoken' \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)